### PR TITLE
Add scopes to the address models for area lookup

### PR DESCRIPTION
**Pull request by cintamani**
From: https://eaflood.atlassian.net/browse/RUBY-487

Add scopes to find addresses missing areas and with valid easting and northing parameters

### CODE REVIEW

File: app/models/waste_exemptions_engine/address.rb [hunk 1]
```diff
@@ -10,5 +10,8 @@ class Address < ActiveRecord::Base
 
     enum address_type: { unknown: 0, operator: 1, contact: 2, site: 3 }
     enum mode: { unknown_mode: 0, lookup: 1, manual: 2, auto: 3 }
+
+    scope :with_easting_and_northing, -> { where.not(x: nil, y: nil) }
+    scope :missing_ea_area, -> { where(area: [nil, ""]) }
```
Comment (Cruikshanks) [inline app/models/waste_exemptions_engine/address.rb]: Just a quick note. I should have picked it up when I reviewed FRAE (I'm happy to sort out my oversight) but I think we should drop the `_ea_` from the scope. I think `:missing_area` is fine hence I'm happy to update FRAE to keep things consistent.

Comment (cintamani) [reply app/models/waste_exemptions_engine/address.rb]: Oh right! I'll make the change!

Comment (cintamani) [inline app/models/waste_exemptions_engine/address.rb]:
```suggestion
    scope :missing_area, -> { where(area: [nil, ""]) }
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 2]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+          create(:address, x: 123.4, y: nil)
+
+          valid_address = create(:address, x: 123.4, y: 123.4)
+
+          expect(described_class.with_easting_and_northing.size).to eq(1)
+          expect(described_class.with_easting_and_northing.first).to eq(valid_address)
+        end
+      end
+
+      describe ".missing_ea_area" do
```
Comment (cintamani) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
      describe ".missing_area" do
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 3]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+        it "returns all address with a missing EA area" do
```
Comment (cintamani) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
        it "returns all address with a missing area" do
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 4]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+          create(:address, area: "West Midlands")
+
+          nil_area = create(:address, area: nil)
+          empty_area = create(:address, area: "")
+
+          expect(described_class.missing_ea_area.size).to eq(2)
```
Comment (cintamani) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
          expect(described_class.missing_area.size).to eq(2)
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 5]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+          expect(described_class.missing_ea_area).to include(empty_area)
```
Comment (cintamani) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
          expect(described_class.missing_area).to include(empty_area)
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 6]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+          expect(described_class.missing_ea_area).to include(nil_area)
```
Comment (cintamani) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
          expect(described_class.missing_area).to include(nil_area)
```

File: spec/models/waste_exemptions_engine/address_spec.rb [hunk 7]
```diff
@@ -16,6 +16,33 @@ module WasteExemptionsEngine
...
+
+          valid_address = create(:address, x: 123.4, y: 123.4)
+
+          expect(described_class.with_easting_and_northing.size).to eq(1)
+          expect(described_class.with_easting_and_northing.first).to eq(valid_address)
+        end
+      end
+
+      describe ".missing_area" do
+        it "returns all address with a missing area" do
```
Comment (Cruikshanks) [inline spec/models/waste_exemptions_engine/address_spec.rb]:
```suggestion
        it "returns all addresses with a missing area" do
```

Comment (cintamani) [reply spec/models/waste_exemptions_engine/address_spec.rb]: Cheers :D

**Pull request merged**